### ISSUE TITLE
Possible issue with UTF-8 locale

### ISSUE THREAD

**Issue by 3id0** [Abjad/abjad#880]
If I run the example from [this page](http://abjad.mbrsi.org/for_beginners/abjad_hello_world_at_the_interpreter.html) in the Python Shell I get the following result:

```
Python 3.6.3 (default, Oct  4 2017, 06:09:05) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from abjad import *
>>> note = Note("c'4")
>>> show(note)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/topleveltools/show.py", line 60, in show
    result = abjad.persist(argument).as_pdf(**keywords)
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/agenttools/PersistenceAgent.py", line 230, in as_pdf
    **keywords
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/agenttools/PersistenceAgent.py", line 74, in as_ly
    illustration = illustrate_function(**keywords)
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/scoretools/Component.py", line 112, in __illustrate__
    lilypond_file = abjad.LilyPondFile.new(self)
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/lilypondfiletools/LilyPondFile.py", line 1098, in new
    use_relative_includes=use_relative_includes,
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/lilypondfiletools/LilyPondFile.py", line 131, in __init__
    token = lilypondfiletools.LilyPondVersionToken()
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/lilypondfiletools/LilyPondVersionToken.py", line 33, in __init__
    version_string = abjad_configuration.get_lilypond_version_string()
  File "/Users/3id0/dev/test-abjad/.venv/lib/python3.6/site-packages/abjad/tools/systemtools/AbjadConfiguration.py", line 213, in get_lilypond_version_string
    encoding = locale.getdefaultlocale()[1]
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/locale.py", line 562, in getdefaultlocale
    return _parse_localename(localename)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/locale.py", line 490, in _parse_localename
    raise ValueError('unknown locale: %s' % localename)
ValueError: unknown locale: UTF-8
>>> 
```

But if I set the following locales before starting the Python Shell and issuing the same commands I don't get any error and the pdf is generated as excepted:
```
$ export LC_ALL=en_US.UTF-8
$ export LANG=en_US.UTF-8
```

I initially installed macOS in french and switched to english down the road, maybe this has something to do with this error.

In any case, thanks for the great module and documentation.

**Comment by josiah-wolf-oberholtzer** [comment]
Thanks for the report. I expect this issue will be fixed in the upcoming version of Abjad, because we're dropping support for Python 2.x in favor of 3.x-only.

Try installing Abjad from the `master` branch - the code related to `get_lilypond_version_string()` has already been changed there.

**Comment by trevorbaca** [comment]
Hi @3id0, does this issue still appear under Abjad 3.0?

**Comment by nagymusic** [comment]
I just came across this post as I tried to solve a similar problem on a Windows machine. If I compile this example code:
```
import abjad
note = abjad.Note()
abjad.show(note)
```
I get the following error message:
```
Python 3.7.2 (tags/v3.7.2:9a3ffc0492, Dec 23 2018, 22:20:52) [MSC v.1916 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import abjad
>>> note = abjad.Note()
>>> abjad.show(note)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\top\show.py", line 51, in show
    result = abjad.persist(argument).as_pdf(**keywords)
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\system\PersistenceManager.py", line 234, in as_pdf
    **keywords
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\system\PersistenceManager.py", line 67, in as_ly
    lilypond_file = illustrate_function(**keywords)
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\core\Component.py", line 118, in __illustrate__
    lilypond_file = abjad.LilyPondFile.new(self)
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\lilypondfile\LilyPondFile.py", line 990, in new
    use_relative_includes=use_relative_includes,
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\lilypondfile\LilyPondFile.py", line 147, in __init__
    token = LilyPondVersionToken()
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\lilypondfile\LilyPondVersionToken.py", line 27, in __init__
    version_string = abjad_configuration.get_lilypond_version_string()
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\site-packages\abjad\system\AbjadConfiguration.py", line 412, in get_lilypond_version_string
    proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "C:\Users\nagym\AppData\Local\Programs\Python\Python37-32\lib\subprocess.py", line 1178, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] The system cannot find the file specified
```
Perhaps it's worth noting that Abjad 3.0.0 and Lilypond 2.19.82 are callable from the terminal. Lastly, ```abjad.f()``` is also returning the Lilypond content:
```
>>> abjad.f(note)
c'4
```
Lastly, this issue persists regardless if Abjad is installed directly via pip (stable) or GitHub (dev). I also set the UTF locales as suggested by 3id0 above but that didn't help. The issue developed only recently as I had to reinstall Abjad on my PC laptop. Before then, Abjad 3.0.0 (installed in late Summer/early Fall 2018) worked just fine.

I welcome your advice on how to go around this. Thank you for yoru amazing work to make Abjad available!

**Comment by nagymusic** [comment]
Resetting the PC solved the problem. I'm still not sure what was the cause of this issue and why Python didn't want to compile the package earlier.

I appreciate the opportunity to ask the question. While I'm mainly using Abjad on Linux and Mac machines, I'm happy to confirm the Abjad works perfectly on Windows 10 systems. 

Thank you!

**Comment by 3id0** [comment]
Hey, sorry for the super late response.

I can confirm this is fixed in Abjad 3.0.0 with Python 3.7.3 on macOS.

Thanks a lot folks!

---

Just in case this is helpful, I got the following warning with this new version, but nothing that prevents me from using `show(note)`:
```
>>> from abjad import *
>>> note = Note("c'4")
WARNING: yacc table file version is out of date
```

For reference (not sure if related):
```
>>> import ply
>>> ply.__version__
'3.11'
```
